pkg/api/pod/warnings_test.go

```diff
@@ -1058,6 +1059,229 @@ func TestWarnings(t *testing.T) {
 				`spec.containers[1].ports[1]: duplicate port definition with spec.containers[1].ports[0]`,
 			},
 		},
+		{
+			name: "create duplicate container ports name in two containers",
+			template: &api.PodTemplateSpec{Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name: "foo1",
+						Ports: []api.ContainerPort{
+							{ContainerPort: 80, HostPort: 80, Protocol: api.ProtocolUDP, Name: "test"},
+							{ContainerPort: 180, HostPort: 180, Protocol: api.ProtocolUDP},
+							{ContainerPort: 8080, HostPort: 8080, Protocol: api.ProtocolUDP, Name: "test"},
```

Comment:
this should not pass validation (dup port within a container) - we should only be testing here things that will pass validation, but need a warning.  This will help keep each testcase focused.